packet/member.py

```diff
@@ -1,11 +1,23 @@
-from .models import Freshman, FreshSignature, UpperSignature, MiscSignature
+from datetime import datetime
+
+from .models import Packet, Freshman, FreshSignature, UpperSignature, MiscSignature, db
 
 
 def signed_packets(member):
     # Checks whether or not member is a freshman
-    if Freshman.query.filter_by(rit_username=member).first() is not None:
-        return FreshSignature.query.filter_by(freshman_username=member, signed=True).all()
+    if db.session.query(FreshSignature.freshman_username) \
+            .filter(FreshSignature.freshman_username == member).count() > 0:
+        return db.session.query(Packet.freshman_username, Freshman.name, FreshSignature.signed) \
+            .select_from(FreshSignature).join(Packet).join(Freshman) \
+            .filter(FreshSignature.freshman_username == member) \
+            .filter(Packet.end > datetime.now()).filter(Packet.start < datetime.now()).all()
     # Checks whether or not member is an upperclassman
-    if UpperSignature.query.filter_by(member=member).first() is not None:
-        return UpperSignature.query.filter_by(member=member, signed=True).all()
-    return MiscSignature.query.filter_by(member=member).all()
+    if db.session.query(UpperSignature.member).join(Packet).join(Freshman) \
+            .filter(UpperSignature.member == member).count() > 0:
+        return db.session.query(Freshman.rit_username, Freshman.name, UpperSignature.signed) \
+            .select_from(UpperSignature).join(Packet).join(Freshman) \
+            .filter(UpperSignature.member == member) \
+            .filter(Packet.end > datetime.now()).filter(Packet.start < datetime.now()).all()
+    return db.session.query(Freshman.rit_username, Freshman.name, MiscSignature.member.isnot(None)) \
```

Comment:
The way it's implemented in the routes, a list of freshmen are required.  For alumni/off floors we do not have a way to check if they've signed a packet except with existence of their signature.  isnot(None) is True if they've signed it, False for not having signed it.  Otherwise it would only returned signed packets which would not give you a list of others to sign